src/main/kotlin/uk/gov/gdx/datashare/services/PrisonerEventMessageProcessor.kt

```diff
@@ -0,0 +1,86 @@
+package uk.gov.gdx.datashare.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.jms.annotation.JmsListener
+import org.springframework.stereotype.Service
+import uk.gov.gdx.datashare.enums.EventType
+import uk.gov.gdx.datashare.models.EventToPublish
+import uk.gov.gdx.datashare.repositories.SupplierSubscriptionRepository
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+@Service
+@ConditionalOnProperty(name = ["api.base.prisoner-event.enabled"], havingValue = "true")
+class PrisonerEventMessageProcessor(
+  private val objectMapper: ObjectMapper,
+  private val dataReceiverService: DataReceiverService,
+  private val supplierSubscriptionRepository: SupplierSubscriptionRepository,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @JmsListener(destination = "prisoner-event", containerFactory = "awsQueueContainerFactoryProxy")
+  fun onPrisonerEventMessage(prisonerEventMessage: String) {
+    val (message, messageAttributes) = objectMapper.readValue(prisonerEventMessage, TopicMessage::class.java)
+    val eventType = messageAttributes.eventType.Value
+    log.info("Received message $message, type $eventType")
+
+    val hmppsDomainEvent = objectMapper.readValue(message, HMPPSDomainEvent::class.java)
+    when (eventType) {
+      "prison-offender-events.prisoner.received" -> processHMPPSEvent(hmppsDomainEvent)
+      else -> {
+        log.debug("Ignoring message with type $eventType")
+      }
+    }
+  }
+
+  fun processHMPPSEvent(hmppsDomainEvent: HMPPSDomainEvent) {
+    // find a supplier with a defined name
+    supplierSubscriptionRepository.findAllByEventType(EventType.ENTERED_PRISON)
```

Comment:
Just find any supplier who supports this event type

Reply:
is this for enrichment? Or just a check that the queue is only running if we have set up HMPPS as a supplier?

Reply:
Just a check that we have a supplier that handles that event type

Reply:
Perhaps a premature optimisation, but we could use `findFirstByEventType` here